### ISSUE TITLE
Slight jump when pinning

### ISSUE THREAD

**Issue by artildo** [janpaepke/ScrollMagic#976]
Hello! When I scroll til the pin, my animation shatters a bit, like it is scrolled more than needed and it jumps back to the pin.
Please see the video (the top of the map)
https://youtu.be/3EaktDx3N_M

What can I do to just fix the animation when it reaches the pin?

**Comment by BB-000** [comment]
+1000 

Spent hours coding a complicated page with multiple pinning elements and it looks awful (unless you scroll very very slowly) :/ 

It's like the scroll event is being throttled? Is there a way to stop this??

I got briefly excited when I found 'refreshInterval' but it doesn't do what I initially thought.

**Comment by artildo** [comment]
Glad to see it's not my persinal issue. Funny, but I used the same themplate I created before, just changed backround, JSONs, function. Nothing changed that could affect scrollmagic. But in the first projects there were no jumps at all

**Comment by artildo** [comment]
I found out that in my first project there were some background styoling. When I removed it, the jumps occured. I had to return it. So I made the "gradient" with equal colors for no gradient effect in real. This saved me.

` background: -moz-linear-gradient(139deg, rgba(42, 73, 142, 1) 0%, rgba(42, 73, 142, 1) 54%, rgba(42, 73, 142, 1) 100%);
  background: -webkit-linear-gradient(139deg, rgba(42, 73, 142, 1) 0%, rgba(42, 73, 142, 1) 54%, rgba(42, 73, 142, 1) 100%);
  background: linear-gradient(139deg, rgba(42, 73, 142, 1) 0%, rgba(42, 73, 142, 1) 54%, rgba(42, 73, 142, 1) 100%);
  filter: progid:DXImageTransform.Microsoft.gradient(startColorstr="#003d77",endColorstr="#005bae",GradientType=1);
  -webkit-box-shadow: 8px 0px 21px 7px rgba(0,0,0,0.16); `

**Comment by sharkfin009** [comment]
Ah, good catch!
So perhaps the gradient was causing repaints which where challenging the
browser?

**Comment by artildo** [comment]
@sharkfin009 

Yeah, this gradient stuff fixed the issue, no jumps with it. Have no idea how it works, kind of magic (scrollmagic)

**Comment by cytronn** [comment]
> +1000
> 
> Yes there is a noticeable delay in the time it takes to pin, causing elements to jump back to the pin position after it has already been scrolled halfway up the page.
> 
> Spent hours coding a complicated page with multiple pinning elements and it looks awful (unless you scroll very very slowly) :/
> 
> It's like the scroll event is being throttled? Is there a way to stop this??
> 
> I got briefly excited when I found 'refreshInterval' but it doesn't do what I initially thought...

Did you find a way to make your project work smoothly?

**Comment by BB-000** [comment]
@cytronn   The new **[GSAP Scroll Trigger plugin](https://greensock.com/scrolltrigger/)** is awesome.

Check out the 'anticipatePin' property [in the docs here](https://greensock.com/docs/v3/Plugins/ScrollTrigger)

"If you pin large sections/panels you may notice what looks like a slight delay in pinning when you scroll quickly. That's caused by the fact that most modern browsers handle scroll repaints on a separate thread, so at the moment of pinning the browser may have already painted the pre-pinned content, making it visible for perhaps 1/60th of a second. The only way to counteract that is to have ScrollTrigger monitor the scroll velocity and anticipate the pin, applying it slightly early to avoid that flash of unpinned content."

**Comment by BB-000** [comment]
@cytronn But no, I could not get my page to be smooth using ScrollMagic 😢

**Comment by cytronn** [comment]
So you rebuilt it with scrolltrigger, is that right?

**Comment by janpaepke** [comment]
A few years late on this one, sorry about that. The jump-on-pin behavior was a fundamental limitation of v2's pinning approach — toggling between `position: relative` and `position: fixed` via JavaScript on a scroll event inherently races against the browser's scroll-thread repaint. No amount of configuration could fully eliminate that 1-frame flash.

v3 does not include a pin system at all. For pinning, CSS `position: sticky` is now the right tool — it is handled natively by the browser's compositor, so there is no JavaScript timing gap and no jitter.

---
**ScrollMagic v3 is now available** — a complete rewrite with an entirely new architecture and API. v2 will not receive further updates (2.0.9 is the final release).

👉 `npm install scrollmagic@next` — [Docs](https://github.com/janpaepke/ScrollMagic/blob/main/README.md) · [Release notes](https://github.com/janpaepke/ScrollMagic/releases/tag/v3.0.0-beta.0) · [npm](https://www.npmjs.com/package/scrollmagic/v/3.0.0-beta.0)
